src/scorer.ts

```diff
@@ -7,42 +7,47 @@ export class Scorer {
   p2Cards: CardInstance[];
   p3Cards: CardInstance[];
   p4Cards: CardInstance[];
+  playersCards: Array<Array<CardInstance>>
 
   constructor(p1Cards: string, p2Cards: string, p3Cards: string, p4Cards: string) {
-    this.p1Cards = p1Cards.split(',').filter(a => a).map((name) => getCard(name));
-    this.p2Cards = p2Cards.split(',').filter(a => a).map((name) => getCard(name));
-    this.p3Cards = p3Cards.split(',').filter(a => a).map((name) => getCard(name));
-    this.p4Cards = p4Cards.split(',').filter(a => a).map((name) => getCard(name));
+    this.playersCards = [
+      this.p1Cards = this.buildCardInsts(p1Cards),
+      this.p2Cards = this.buildCardInsts(p2Cards),
+      this.p3Cards = this.buildCardInsts(p3Cards),
+      this.p4Cards = this.buildCardInsts(p4Cards)
+      ]
+
   }
 
   // Functions
-  scoreA(): number[] {
-    return [
-      this.p1Cards.reduce((total, inst) => total + inst.card.pointsA, 0),
-      this.p2Cards.reduce((total, inst) => total + inst.card.pointsA, 0),
-      this.p3Cards.reduce((total, inst) => total + inst.card.pointsA, 0),
-      this.p4Cards.reduce((total, inst) => total + inst.card.pointsA, 0),
-    ]
-  }
+  scores(): number[] {
+    this.playersCards.forEach((playerCards) => {
+      playerCards.forEach((inst: CardInstance) => {
+        if(typeof inst.card.pointsA === 'function') {
+          inst.finalA = inst.card.pointsA(inst)
+        } else {
+          inst.finalA = inst.card.pointsA
+        }
```

Comment:
Rather than doing this any reason that you can't have it always be a function? The function can return a metadata variable, or even just be an inline function that returns a value.